tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -490,11 +490,11 @@ def test_structured_dataset_in_dataclass():
     df = pd.DataFrame({"Name": ["Tom", "Joseph"], "Age": [20, 22]})
 
     @dataclass
-    class InnerDatasetStruct(DataClassJsonMixin):
```

Comment:
I think we should keep the `DataClassJsonMixin` to make sure flytekit can still serialize this dataclass.

Reply:
No problem